### PR TITLE
[Feature] sc-26163 Ask user to fill credentials in non dbt case

### PR DESCRIPTION
**Pull request by kentwelcome**
- Refactor user input function by inquirer
 - Refactor DataSource field to support default value and optional
 - Refactor workspace to extract DataSource and Configuration to other
   files

Signed-off-by: Kent Huang <kent@infuseai.io>

### CODE REVIEW

**Comment by kentwelcome** [comment]
CleanShot 2022-06-07 at 00 54 09

**Comment by kentwelcome** [comment]
CleanShot 2022-06-07 at 00 55 35

File: piperider_cli/workspace.py [hunk 1]
```diff
@@ -588,7 +274,10 @@ def search_dbt_project_path():
 
 
 def init(dbt_project_path=None, dbt_profiles_dir=None):
-    _generate_piperider_workspace()
+    console = Console()
+    if _generate_piperider_workspace() is False:
+        console.print(f'[bold green]Piperinfuseider workspace already exist[/bold green] ')
```
Comment (qrtt1) [inline piperider_cli/workspace.py:279]: what is `Piperinfuseider`?

Comment (kentwelcome) [reply piperider_cli/workspace.py:279]: Miss typing. Orz

Comment (kentwelcome) [reply piperider_cli/workspace.py]: Fix it in the new commit

File: piperider_cli/datasource.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+
+class PostgreSQLDataSource(DataSource):
+    def __init__(self, name, **kwargs):
+        super().__init__(name, 'postgres', **kwargs)
+        self.fields = [
+            TextField('host', description='Host URL'),
+            NumberField('port', default=5432, description='Port'),
+            TextField('user', description='Username'),
+            PasswordField('password', description='Password (hidden)'),
```
Comment (ctiml) [inline piperider_cli/datasource.py:190]: We can remove ` (hidden)` since user input is masked with asterisk

Comment (kentwelcome) [reply piperider_cli/datasource.py]: fixed!

**Comment by kentwelcome** [comment]
For the case of multiple data sources, will ask user to select a data source to init.
CleanShot 2022-06-07 at 11 22 46

**Pull request merged**